core/src/main/scala/zio/logging/Logging.scala

```diff
@@ -1,45 +1,95 @@
 package zio.logging
 
+import zio._
 import zio.clock.{ currentDateTime, Clock }
 import zio.console.{ putStrLn, Console }
-import zio._
 
 object Logging {
-  type Logging = Has[Logging.Service]
-
-  trait Service {
-    def logger: Logger
-  }
+  type Logging = Has[Logger[String]]
 
   def console(
-    format: (LogContext, => String) => String
+    format: (LogContext, => String) => String,
+    rootLoggerName: Option[String] = None
   ): ZLayer[Console with Clock, Nothing, Logging] =
-    make((context, line) =>
-      for {
-        date       <- currentDateTime.orDie
-        level      = context(LogAnnotation.Level)
-        loggerName = context(LogAnnotation.Name)
-        maybeError = context
-          .get(LogAnnotation.Throwable)
-          .map(Cause.fail)
-          .orElse(context.get(LogAnnotation.Cause))
-          .map(cause => System.lineSeparator() + cause.prettyPrint)
-          .getOrElse("")
-        _ <- putStrLn(date.toString + " " + level + " " + loggerName + " " + format(context, line) + " " + maybeError)
-      } yield ()
+    make(
+      (context, line) =>
+        for {
+          date       <- currentDateTime.orDie
+          level      = context(LogAnnotation.Level)
+          loggerName = context(LogAnnotation.Name)
+          maybeError = context
+            .get(LogAnnotation.Throwable)
+            .map(Cause.fail)
+            .orElse(context.get(LogAnnotation.Cause))
+            .map(cause => System.lineSeparator() + cause.prettyPrint)
+            .getOrElse("")
+          _ <- putStrLn(date.toString + " " + level + " " + loggerName + " " + format(context, line) + " " + maybeError)
+        } yield (),
+      rootLoggerName
     )
 
+  val context: URIO[Logging, LogContext] =
+    ZIO.accessM[Logging](_.get.logContext)
+
+  def debug(line: => String): ZIO[Logging, Nothing, Unit] =
```

Comment:
I think these can be polymorphic without any loss in ergonomics for the String case

Reply:
I will keep this for now. We need to think about "Structured Logging" probably via `Message` from LogStage. then this will be probably polymorphic...

Reply:
I think for LogStage, we can have:

```scala
package object logstage {
  type LogStage = Has[Logger[Message]]
  object LogStage {
    // helpers
  }
}
```

Reply:
there's already full layer support in logstage - https://izumi.7mind.io/latest/release/doc/logstage/index.html#zio-environment-support
It's a matter of just re-exporting if you want to add it here, I guess

Reply:
thanks @neko-kai I will open separate PR for that.